### PR TITLE
Formalize adding a "next" release to metainfo.xml

### PR DESCRIPTION
**Pull request by dylanmccall**
This is my attempt to clean up how we do release notes in this project.

First, I'm building on #114, making sure the About window shows release notes for the next release. We know what the next release is because its name is always "NEXT" (instead of trying to guess at what its number will be).

Next, I'm adding a little bit to the `bump-my-version` (previously `bump2version`) config so, when we create a release, it finds that `<release version="NEXT" …>` entry in metainfo.xml (via a blob of regex) and rewrites it with the new release ID, `type="stable"`, and the current date.

And I'm changing the README to explain what to do here. I think it's much less tedious than it was, since one only needs to touch the metainfo file when they're actually making changes.

This could do with a little bit more cleverness, such as a pre-commit hook that yells at you if you don't have a "NEXT" version entry, but unfortunately there isn't a convenient "XML has an element matching this selector" hook at https://pre-commit.com/hooks.html, so it would involve more work than I should be putting into this right now.

But I think it's a good start and should serve us well here :)

### CODE REVIEW

**Comment by dylanmccall** [comment]
Oh, annoying, `appstream-util` has a better idea what order releases should go in, so it looks like I'll have to change the format to either `version="99999"` or `version="0.7+next`. I'll go with the latter since it feels more useful.

File: .bumpversion.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+]
+
+[[tool.bumpversion.files]]
+filename = "VERSION"
+
+[[tool.bumpversion.files]]
+filename = "data/metainfo/org.endlessos.Key.metainfo.xml.in.in"
+# This crude regex will break if the release tag contains any "/" or ">" characters.
+search = "<release\\s([^>]+\\s)*version=\"{current_version}\\+next\"[^\\/>]*"
+replace = "<release version=\"{new_version}\" date=\"{now:%Y-%m-%d}\" type=\"stable\""
```
Comment (dylanmccall) [inline .bumpversion.toml:18]: The point here is to support self-closing `<release />` tags as well as `<release>…</release>` pairs. It's ugly, and I think there's probably a better regex one could do here, but this felt the _simplest_ that fits within the current specification. (Hopefully if we ever need to change that, someone will have done something clever so we can use a real XML parser here).

File: .bumpversion.toml [hunk 2]
```diff
@@ -4,9 +4,16 @@ commit = true
...
 parse = "(?P<major>\\d+)\\.(?P<minor>\\d+)(.(?P<patch>\\d+))?"
 serialize = [
-    "{major}.{minor}.{patch}"
+    "{major}.{minor}.{patch}",
     "{major}.{minor}"
 ]
 
 [[tool.bumpversion.files]]
-filename = VERSION
+filename = "VERSION"
```
Comment (dbnicholson) [inline .bumpversion.toml:12]: This change and above should obviously be part of the previous commit.

**Review by dbnicholson — APPROVED** [review]
Seems pretty nice! You'll need to rebase, though.

**Pull request merged**